arwen/elrondapi/elrondei.go

```diff
@@ -1657,24 +1673,23 @@ func executeReadOnly(
 	gasToUse = metering.GasSchedule().BaseOperationCost.DataCopyPerByte * uint64(actualLen)
 	metering.UseGas(gasToUse)
 
-	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+	if err != nil {
```

Comment:
arwen.WithFault must remain.  please make the changes back. the execution must stop in these errors. We want to have the protected arwen, in which any error really stops the execution.

Reply:
Restored.